File: python-server/src/tools/add_todo.py
from ..types import Todo, todos
import uuid


async def add_todo(title: str, description: str | None = None) -> str:
    todo_id = str(uuid.uuid4())
    todos[todo_id] = Todo(id=todo_id, title=title, description=description, completed=False)

    return f"Added todo: {title} (ID: {todo_id})"

File: ts-server/src/tools/add-todo.ts
import { todos } from "../types.js";

export async function addTodo(title: string, description?: string) {
  const id = crypto.randomUUID();
  todos[id] = { id, title, description, completed: false };

  return {
    content: [
      {
        type: "text" as const,
        text: `Added todo: ${title} (ID: ${id})`,
      },
    ],
  };
}
